packages/medusa-file-s3/src/services/s3.js

```diff
@@ -3,28 +3,23 @@ import aws from "aws-sdk"
 import { AbstractFileService } from '@medusajs/medusa'
 
 class S3Service extends AbstractFileService {
-  constructor({}, options) {
+  constructor({ }, options) {
     super({}, options)
 
     this.bucket_ = options.bucket
-    this.s3Url_ = options.s3_url
-    this.accessKeyId_ = options.access_key_id
-    this.secretAccessKey_ = options.secret_access_key
-    this.region_ = options.region
-    this.endpoint_ = options.endpoint
+
+    // Remove parameters not supported by aws-sdk
+    delete options.bucket
+
+    this.awsOptions_ = options
   }
```

Comment:
todo: we need this to remain backward compatible for anyone who might be using the plugin in its current state - can I ask you to add a "mapper" that takes the existing options and maps them to the correct corresponding awsOptions?

Reply:
@srindom Are you on board with the latest changes?